src/components/PopupConfirm.js

```diff
@@ -0,0 +1,26 @@
+import Popup from "./Popup.js";
+
+export default class PopupConfirm extends Popup {
+  constructor(popupSelector, handleFormSubmit) {
+    super(popupSelector);
+    this._popupForm = this._popup.querySelector(".popup__form");
+    this._handleFormSubmit = handleFormSubmit;
+    this._buttonText = this._popupForm.querySelector(".popup__button");
+    this.setEventListeners();
+  }
+
+  deleting(show) {
+    show ?
+      this._buttonText.textContent = "Deleting..." :
+      this._buttonText.textContent = "Yes";
+  }
+
+  setEventListeners = () => {
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleFormSubmit();
+      evt.target.reset();
```

Comment:
Please, delete `evt.target.reset();` from `PopupConfirm` and `PopupWithForm`

You need to reset the form only in block `then` after a successful response from the server because if there is an error the user should be able to send the request again without having to type all the info again.